### PR TITLE
Support for deregister entity tx

### LINKED ISSUE #135: Support for Registry Deregister Entity txn type
## Steps to reproduce
```bash
cd $(mktemp -d)

OASIS_CORE_VERSION=21.2.7
wget https://github.com/oasisprotocol/oasis-core/releases/download/v${OASIS_CORE_VERSION}/oasis_core_${OASIS_CORE_VERSION}_linux_amd64.tar.gz
tar -xf oasis_core_${OASIS_CORE_VERSION}_linux_amd64.tar.gz
export OASIS_NODE="$PWD/oasis_core_${OASIS_CORE_VERSION}_linux_amd64/oasis-node"

OASIS_CORE_LEDGER_VERSION=1.2.0
wget https://github.com/oasisprotocol/oasis-core-ledger/releases/download/v${OASIS_CORE_LEDGER_VERSION}/oasis_core_ledger_${OASIS_CORE_LEDGER_VERSION}_linux_amd64.tar.gz
tar -xf oasis_core_ledger_${OASIS_CORE_LEDGER_VERSION}_linux_amd64.tar.gz
export LEDGER_SIGNER_PATH="$PWD/oasis_core_ledger_${OASIS_CORE_LEDGER_VERSION}_linux_amd64/ledger-signer"

wget https://github.com/oasisprotocol/mainnet-artifacts/releases/download/2021-04-28/genesis.json -O genesis.json

ENTITY_DIR=entity0/
LEDGER_SIGNER_FLAGS=(
  --signer.dir $ENTITY_DIR
  --signer.backend plugin
  --signer.plugin.name ledger
  --signer.plugin.path "$LEDGER_SIGNER_PATH"
)
TX_FLAGS=(
  --genesis.file genesis.json
  "${LEDGER_SIGNER_FLAGS[@]}"
)

# Open Oasis app on Ledger.

mkdir $ENTITY_DIR
$OASIS_NODE signer export "${LEDGER_SIGNER_FLAGS[@]}"

$OASIS_NODE registry entity gen_deregister \
  "${TX_FLAGS[@]}" \
  --transaction.file tx_entity_registery.json \
  --transaction.nonce 42 \
  --transaction.fee.gas 1000 \
  --transaction.fee.amount 2000
```

## Results with Version 2.2.0

NOTE: I've compiled the latest commit of PR #129 as of now: 551df7d2d3f7df0ec8fe3099ad6d859bb1dac4e7.

```
$ $OASIS_NODE registry entity gen_deregister \
>   "${TX_FLAGS[@]}" \
>   --transaction.file tx_entity_registery.json \
>   --transaction.nonce 42 \
>   --transaction.fee.gas 1000 \
>   --transaction.fee.amount 2000
You are about to sign the following transaction:
  Method: registry.DeregisterEntity
  Body:
    <unknown method body: >
  Nonce:  42
  Fee:
    Amount: 0.000002 ROSE
    Gas limit: 1000
    (gas price: 0.000000002 ROSE per gas unit)
Other info:
  Genesis document's hash: 53852332637bacb61b91b6411ab4095168ba02a50be4c3f82448438826f23898

You may need to review the transaction on your device if you use a hardware-based signer plugin...
ts=2021-08-03T19:39:25.384414697Z level=error module=cmd/common/consensus caller=consensus.go:139 msg="failed to sign transaction" err="signature/signer/plugin: failed to sign: ledger: failed to sign message: ledger/oasis: failed to sign: No more data"
```

### PR DESCRIPTION
**Pull request by emmanuelm41**
Closes #135

### CODE REVIEW

File: app/src/consumer/parser_consumer.c [hunk 1]
```diff
@@ -709,9 +709,26 @@ __Z_INLINE parser_error_t parser_getItemTx(const parser_context_t *ctx,
...
+                                                pageCount);
+                }
+                case 2: {
+                    snprintf(outKey, outKeyLen, "Gas limit");
+                    uint64_to_str(outVal, outValLen, parser_tx_obj.oasis.tx.fee_gas);
+                    *pageCount = 1;
+                    return parser_ok;
+                }
+            }
+        }
```
Comment (jleni) [inline app/src/consumer/parser_consumer.c:731]: we need a default case that returns an error

File: app/src/consumer/parser_impl_con.c [hunk 2]
```diff
@@ -563,7 +563,7 @@ __Z_INLINE parser_error_t _readBody(parser_tx_t *v, CborValue *rootItem) {
             CHECK_CBOR_ERR(cbor_value_advance(&contents))
 
             break;
-        }
+        }/*
         case registryDeregisterEntity: {
```
Comment (jleni) [inline app/src/consumer/parser_impl_con.c:567]: This seems like obsolete code. 
Can we remove it?

**Review by jleni — CHANGES_REQUESTED** [review]

File: app/src/consumer/parser_consumer.c (hunk 1 above)
Comment (emmanuelm41) [reply app/src/consumer/parser_consumer.c:731]: The default case is at the end of the function! https://github.com/Zondax/ledger-oasis/blob/ea3de0a00f54caefe56b38b47148901f328fa15f/app/src/consumer/parser_consumer.c#L910

File: app/src/consumer/parser_impl_con.c (hunk 2 above)
Comment (emmanuelm41) [reply app/src/consumer/parser_impl_con.c:567]: I am waiting @tjanez to confirm that the ui flow is correct. I may be wrong and we still need it.

Comment (tjanez) [reply app/src/consumer/parser_impl_con.c:567]: I'm confirming the workflow is correct and matches https://github.com/Zondax/ledger-oasis/issues/135#issuecomment-893493809.

Comment (emmanuelm41) [reply app/src/consumer/parser_impl_con.c]: Great!

**Review by jleni — APPROVED** [review]

**Pull request merged**